ibis/backends/tests/test_array.py

```diff
@@ -803,10 +803,13 @@ def test_array_intersect(con, data):
 )
 @pytest.mark.notimpl(["postgres"], raises=PsycoPg2SyntaxError)
 @pytest.mark.notimpl(["risingwave"], raises=PsycoPg2InternalError)
-@pytest.mark.notimpl(["datafusion", "flink"], raises=com.OperationNotDefinedError)
+@pytest.mark.notimpl(["datafusion"], raises=com.OperationNotDefinedError)
 @pytest.mark.broken(
     ["trino"], reason="inserting maps into structs doesn't work", raises=TrinoUserError
 )
+@pytest.mark.broken(
+    ["flink"], raises=Py4JJavaError, reason="error on memtable compilation"
```

Comment:
It seems like we should avoid introducing a `broken` marker here by fixing the memtable compilation in this PR. Otherwise, we're potentially masking a bug in the unnest code generation.

Reply:
I tried to fix memtable compilation. Separate from the broken memtable compilation the three tests that I marked as broken also do not work because Flink doesn't have support for them (specifically, Flink doesn't support creating empty array literals and creating named structs with a single field; I found an open JIRA ticket for the former and submitted a JIRA ticket for the latter). I updated the failure reason for these three tests and will open another PR for memtable compilation.